File: docs/h5/src/world/Tile.js
export const TILE_TYPE = {
    WALL: 0,
    FLOOR: 1,
    DOOR: 2,
    STAIRS_DOWN: 3
};

export class Tile {
    constructor(type = TILE_TYPE.WALL) {
        this.type = type;
        this.visible = false; // Fog of War
        this.discovered = false;
        this.walkable = type !== TILE_TYPE.WALL;
    }
}

File: core/src/main/java/com/goldsprite/magicdungeon/world/Tile.java
package com.goldsprite.magicdungeon.world;

public class Tile {
	public TileType type;
	public boolean visible;
	public boolean discovered;
	public boolean walkable;

	public Tile(TileType type) {
		this.type = type;
		this.visible = false;
		this.discovered = false;
		// Walkable logic
		this.walkable = (type != TileType.Wall && type != TileType.Pillar && type != TileType.Torch && type != TileType.Window);
	}
}
